src/utilities/urlExtension.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * Returns the file extension of the path component of a URL
+ * @param {string} url
+ * @returns {string} null if no extension found
+ */
+export function getUrlExtension( url ) {
+
+	let parsedUrl;
+	try {
+
+		parsedUrl = new URL( url );
+
+	} catch ( _ ) {
+
+		try {
+
+			// Try again as a relative URL
+			parsedUrl = new URL( 'https://example.com/' + url );
+
+		} catch ( _ ) {
+
+			// Ignore invalid URLs
+			return null;
+
+		}
+
+	}
```

Comment:
You can use the second argument of `URL` to specify the base url which will be used as a fallback if needed for the passed in URL stem so we won't have to use two try / catches:

```js
parsedUrl = new URL( url, 'http://fakehost.com/' );
```

Reply:
Today I learned:

>  If url is an absolute URL, a given base will be ignored.

https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#parameters